libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/LocationEngineWrapper.kt

```diff
@@ -0,0 +1,79 @@
+package com.mapbox.navigation.core.trip.session
+
+import android.annotation.SuppressLint
+import android.app.PendingIntent
+import android.os.Looper
+import com.mapbox.android.core.location.LocationEngine
+import com.mapbox.android.core.location.LocationEngineCallback
+import com.mapbox.android.core.location.LocationEngineRequest
+import com.mapbox.android.core.location.LocationEngineResult
+import java.lang.Exception
+
+/**
+ * LocationEngine does not have a method to cancel get last location task,
+ * as it has `removeLocationUpdates` for `requestLocationUpdates`.
+ * This may lead to getLastLocation callback firing after `stopLocationUpdates` was invoked,
+ * which is not expected behaviour in context of Mapbox trip sessions.
+ * This class introduces an ability to cancel get last location task for a particular callback.
+ * The issue is reproducible with `ReplayLocationTest#replay_session_locations_do_not_contain_locations_from_previous_session`.
+ */
+internal class LocationEngineWrapper(
```

Comment:
Have you already opened ticket for core to add ability to cancel get last location? I think it would be nice to mention the ticket in the comment.

Reply:
It will be a breaking change. And it would not cause a lot of problems IRL. 
Created one with low priority: https://mapbox.atlassian.net/browse/CORESDK-1819

Reply:
Great! Do you think it makes sense to mention in the comment that `LocationEngineWrapper` exists because we don't have cancellable API and can be removed when CORESDK-1819 is completed ?

Reply:
It's already mentioned. Added the ticket number.